npm/angular/src/mount.ts

```diff
@@ -0,0 +1,211 @@
+ * @param {TestBedConfig} config TestBed configuration passed into the mount function
+ * @returns {TestBed} TestBed
+ */
+function initTestBed<T extends object> (
+  component: Type<T>,
+  config: TestBedConfig<T>,
+): TestBed {
+  const { providers, ...configRest } = config
+
+  const testBed: TestBed = getTestBed()
+
+  testBed.resetTestEnvironment()
+
+  testBed.initTestEnvironment(
+    BrowserDynamicTestingModule,
+    platformBrowserDynamicTesting(),
+    {
+      teardown: { destroyAfterEach: false },
+    },
+  )
+
+  testBed.configureTestingModule({
+    ...bootstrapModule(component, configRest),
+  })
+
+  if (providers != null) {
+    testBed.overrideComponent(component, {
+      add: {
+        providers,
+      },
+    })
+  }
+
+  return testBed
+}
+
+/**
+ * Creates the ComponentFixture
+ *
+ * @param component Angular component being mounted
+ * @param testBed TestBed
+ * @param autoDetectChanges boolean flag defaulted to true that turns on change detection automatically
+ * @returns {ComponentFixture<T>} ComponentFixture
+ */
+function setupFixture<T extends object> (
+  component: Type<T>,
+  testBed: TestBed,
+  autoDetectChanges: boolean,
+): ComponentFixture<T> {
+  const fixture = testBed.createComponent(component)
+
+  fixture.whenStable().then(() => {
+    fixture.autoDetectChanges(autoDetectChanges)
+  })
+
+  return fixture
+}
+
+/**
+ * Gets the componentInstance and Object.assigns any inputs() passed in the TestBedConfig
+ *
+ * @param {TestBedConfig} config TestBed configuration passed into the mount function
+ * @param {ComponentFixture<T>} fixture Fixture for debugging and testing a component.
+ * @returns {T} Component being mounted
+ */
+function setupComponent<T extends object> (
+  config: TestBedConfig<T>,
+  fixture: ComponentFixture<T>,
+): T {
+  let component: T = fixture.componentInstance
+
+  if (config?.inputs) {
+    component = Object.assign(component, config.inputs)
+  }
+
+  return component
+}
+
+/**
+ * Mounts an Angular component inside Cypress browser
+ *
+ * @param {Type<T>} component imported from angular file
+ * @param {TestBedConfig<T>} config configuration used to configure the TestBed
+ * @param {boolean} autoDetectChanges boolean flag defaulted to true that turns on change detection automatically
+ * @example
+ * import { HelloWorldComponent } from 'hello-world/hello-world.component'
+ * import { MyService } from 'services/my.service'
+ * import { SharedModule } from 'shared/shared.module';
+ * import { mount } from '@cypress/angular'
+ * it('can mount', () => {
+ *  mount(HelloWorldComponent, {
+ *    providers: [MyService],
+ *    imports: [SharedModule]
+ *  })
+ *  cy.get('h1').contains('Hello World')
+ * })
+ * @returns Cypress.Chainable<MountResponse<T>>
+ */
+export function mount<T extends object> (
+  component: Type<T>,
+  config: TestBedConfig<T> = {},
+  autoDetectChanges = true,
+): Cypress.Chainable<MountResponse<T>> {
```

Comment:
Our other adapters (react, vue) both accept some additional args that let you inject some stylesheets. I wonder if this is something that would also be useful for angular?

This is implemented using `@cypress/mount-utils`.

React example: https://github.com/cypress-io/cypress/blob/7b58800166f830fd819c55fa91ba1196537928ec/npm/react/src/mount.ts#L4-L10

Reply:
This is a behavior that you cannot do in Angular. You cannot just import a stylesheet at the component level. You have global stylesheets which will be brought in and added to the `webpack` compilation via the users `angular.json` file, and the component themselves stylesheet